packages/web3-eth-iban/src/index.ts

```diff
@@ -0,0 +1,157 @@
+import { toChecksumAddress, isAddress, leftPad, hexToNumber, HexString, InvalidAddressError } from 'web3-utils';
+import { IbanLengthError } from 'web3-common';
+import { IbanOptions } from './types';
+
+/**
+ * Prepare an IBAN for mod 97 computation by moving the first 4 chars to the end and transforming the letters to
+ * numbers (A = 10, B = 11, ..., Z = 35), as specified in ISO13616.
+ */
+const _iso13616Prepare = (iban: string): string => {
+	const A = 'A'.charCodeAt(0);
+	const Z = 'Z'.charCodeAt(0);
+
+	const upperIban = iban.toUpperCase();
+	const modifiedIban = `${upperIban.slice(4)}${upperIban.slice(0, 4)}`;
+
+	return modifiedIban
+		.split('')
+		.map(n => {
+			const code = n.charCodeAt(0);
+			if (code >= A && code <= Z) {
+				// A = 10, B = 11, ... Z = 35
+				return code - A + 10;
+			}
+			return n;
+		})
+		.join('');
+};
+
+/**
+ * return the bigint of the given string with the specified base
+ */
+const _parseInt = (str: string, base: number): bigint =>
+	[...str].reduce((acc, curr) => BigInt(parseInt(curr, base)) + BigInt(base) * acc, 0n);
+
+/**
+ * Calculates the MOD 97 10 of the passed IBAN as specified in ISO7064.
+ */
+const _mod9710 = (iban: string): number => {
+	let remainder = iban;
+	let block;
+
+	while (remainder.length > 2) {
+		block = remainder.slice(0, 9);
+		remainder = `${(parseInt(block, 10) % 97).toString()}${remainder.slice(block.length)}`;
+	}
+
+	return parseInt(remainder, 10) % 97;
+};
+
+const _isValid = (iban: string): boolean =>
+	/^XE[0-9]{2}(ETH[0-9A-Z]{13}|[0-9A-Z]{30,31})$/.test(iban) &&
+	_mod9710(_iso13616Prepare(iban)) === 1;
```

Comment:
Regarding this method and the above un-exported `const`s, should they be private/static methods on the `Iban` class? They're only used in this file for the `Iban` class, so I think it makes sense to have them be at least private methods since that's how they're being used